packages/core/src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -44,12 +52,70 @@ function setup(options: ReactNativeErrorHandlersOptions): void {
  * Setup unhandled promise rejection tracking
  */
 function setupUnhandledRejectionsTracking(patchGlobalPromise: boolean): void {
-  if (patchGlobalPromise) {
-    polyfillPromise();
-  }
+  try {
+    if (isHermesEnabled() && RN_GLOBAL_OBJ.HermesInternal?.enablePromiseRejectionTracker) {
+      logger.log('Using Hermes native promise rejection tracking');
+
+      RN_GLOBAL_OBJ.HermesInternal.enablePromiseRejectionTracker({
+        allRejections: true,
+        onUnhandled: (id: string, error: unknown) => {
+          if (__DEV__) {
+            // eslint-disable-next-line no-console
+            console.warn(`Possible Unhandled Promise Rejection (id: ${id}):\n${error}`);
+          }
+
+          captureException(error, {
+            data: { id },
+            originalException: error,
+            syntheticException: isErrorLike(error) ? undefined : createSyntheticError(),
+            mechanism: { handled: true, type: 'onunhandledrejection' },
+          });
+        },
+        onHandled: (id: string) => {
+          if (__DEV__) {
+            // eslint-disable-next-line no-console
+            console.warn(
+              `Promise Rejection Handled (id: ${id})\n` +
+                'This means you can ignore any previous messages of the form ' +
+                `"Possible Unhandled Promise Rejection (id: ${id})"`,
+            );
+          }
+        },
+      });
+
+      logger.log('Unhandled promise rejections will be caught by Sentry.');
+    } else if (isWeb()) {
+      logger.log('Using Browser JS promise rejection tracking for React Native Web');
+
+      // Use Sentry's built-in global unhandled rejection handler
+      addGlobalUnhandledRejectionInstrumentationHandler((error: unknown) => {
+        captureException(error, {
+          originalException: error,
+          syntheticException: isErrorLike(error) ? undefined : createSyntheticError(),
+          mechanism: { handled: false, type: 'onunhandledrejection' },
+        });
+      });
 
-  attachUnhandledRejectionHandler();
-  checkPromiseAndWarn();
+      logger.warn(
+        'Unhandled promise rejections will not be caught by Sentry. ' +
+          'See https://docs.sentry.io/platforms/react-native/troubleshooting/ for more details.',
+      );
+    } else if (patchGlobalPromise) {
+      // For JSC and other environments, use the existing approach
+      polyfillPromise();
+      attachUnhandledRejectionHandler();
+      checkPromiseAndWarn();
+    } else {
+      // Patching was disabled by user configuration
```

Comment:
how users can disable this configuration on web? it will always enter on the else if on line 87

Reply:
Good point. On the web we always use the [browser rejection tracking](https://github.com/getsentry/sentry-react-native/issues/3957#issuecomment-2262487656). I updated the comment and log to be more clear. Let me know it that makes sense 🙏